src/EnergyPlus/extendedHI.cc

```diff
@@ -0,0 +1,369 @@
+    {
+        Real64 T;
+        std::string region;
+        int SolFla;
+
+        if (eqvar_name == "phi") {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return std::get<1>(find_eqvar(state, T, 1.0)) - eqvar; }, 0.0, 240.0);
+            region = "I";
+        } else if (eqvar_name == "Rf") {
+            General::SolveRoot(
+                state,
+                tol,
+                maxIter,
+                SolFla,
+                T,
+                [&](Real64 T) { return std::get<2>(find_eqvar(state, T, std::min(1.0, Pa0 / pvstar(T)))) - eqvar; },
+                230.0,
+                300.0);
+            region = (Pa0 > pvstar(T)) ? "II" : "III";
+        } else if (eqvar_name == "Rs" || eqvar_name == "Rs*") {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return std::get<3>(find_eqvar(state, T, Pa0 / pvstar(T))) - eqvar; }, 295.0, 350.0);
+            region = (eqvar_name == "Rs") ? "IV" : "V";
+        } else {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return std::get<4>(find_eqvar(state, T, Pa0 / pvstar(T))) - eqvar; }, 340.0, 1000.0);
+            region = "VI";
+        }
+
+        return std::make_pair(T, region);
+    }
+
+    Real64 heatindex(EnergyPlusData &state, Real64 Ta, Real64 RH, bool show_info = false)
+    {
+
+        // Ta: temperature in Kelvin
+        // RH: relative humidity in range of 0.0 to 1.0
+        // show_info: whether to print some messages. Might be useful in debugging
+        // The function computes the extended heat index, in Kelvinn
+
+        auto const HVACSystemRootSolverBackup = state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver;
+        state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver = HVACSystemRootSolverAlgorithm::Bisection;
+        // Dictionary to map eqvar_name to tuple index
+        std::map<std::string, int> dic = {{"phi", 1}, {"Rf", 2}, {"Rs", 3}, {"Rs*", 3}, {"dTcdt", 4}};
+        auto eqvars = find_eqvar(state, Ta, RH);
+        std::string eqvar_name = std::get<0>(eqvars);
+        Real64 eqvar_value;
+        if (eqvar_name == "phi") {
+            eqvar_value = std::get<1>(eqvars);
+        } else if (eqvar_name == "Rf") {
+            eqvar_value = std::get<2>(eqvars);
+        } else if (eqvar_name == "Rs" || eqvar_name == "Rs*") {
+            eqvar_value = std::get<3>(eqvars);
+        } else if (eqvar_name == "dTcdt") {
+            eqvar_value = std::get<4>(eqvars);
+        }
+
+        auto result = find_T(state, eqvar_name, eqvar_value);
+        Real64 T = std::get<0>(result);
+        std::string region = std::get<1>(result);
+
+        if (Ta == 0.0) T = 0.0;
+
+        if (show_info) {
+            if (region == "I") {
+                std::cout << "Region I, covering (variable phi)\n";
```

Comment:
Are these supposed to be debug messages that can be removed?  I dislike seeing std::cout directly in E+ code, even if it's only meant to be used during debugging.